### PR TITLE
initial attempt at task documentation

### PR DESCRIPTION
**Pull request by philvarner**
- https://github.com/stac-utils/stac-task/issues/33

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -6,9 +6,6 @@
 [![codecov](https://codecov.io/gh/stac-utils/stac-task/branch/main/graph/badge.svg)](https://codecov.io/gh/stac-utils/stac-task)
 [![License](https://img.shields.io/badge/License-Apache%202.0-blue.svg)](https://opensource.org/licenses/Apache-2.0)
 
-
-
-
 This Python library consists of the Task class, which is used to create custom tasks based
 on a "STAC In, STAC Out" approach. The Task class acts as wrapper around custom code and provides
```
Comment (stark525) [inline README.md:10]: When I first learned about cirrus and tasks, the biggest confuser was around tasks being "wrappers"; I thought that wrappers meant around the compute (docker or lambda), e.g. this was code that was used to somehow invoke docker containers, not be executed from within a compute context.

I don't know the best place to talk about this; also happy to expand on what I mean.

File: docs/source/usage.rst [hunk 2]
```diff
@@ -12,7 +15,206 @@ To use STAC Task, first install it using pip:
...
+
+   # Support for running as a Lambda Function
+   def handler(event: dict[str, Any], context: Any) -> dict[str, Any]:
+      return MyTask.handler(event)  # type: ignore
+
+   # Support for running as a CLI application
+   if __name__ == "__main__":
+      MyTask.cli()
+
+The expected input looks like the following:
```
Comment (stark525) [inline docs/source/usage.rst:83]: Consider adding a link to ProcessPayload definition, perhaps also identifying what is important and what's not important for local development. I'm not sure the best place for this, but maybe some discussion around
- upload options are only needed if you're uploading to a stac api
- this regex matches all ids and sends it to `my-collection`, and should be used as a default, however you may want to filter specific IDs and send to different collections

Comment (stark525) [reply docs/source/usage.rst:83]: Ack, I just saw lines 99-101. This is probably OBE.

File: docs/source/usage.rst [hunk 3]
```diff
@@ -12,7 +15,206 @@ To use STAC Task, first install it using pip:
...
+---------
+
+To run a Task as a CLI application, add a main definition to the class inheriting Task:
+
+.. code-block:: python
+
+   if __name__ == "__main__":
+      MyTask.cli()
+
+This provides a CLI that supports several useful flags for using stactask. Invoking it
```
Comment (stark525) [inline docs/source/usage.rst:162]: `stactask` -> `stac-task`

File: docs/source/usage.rst [hunk 4]
```diff
@@ -12,7 +15,206 @@ To use STAC Task, first install it using pip:
...
+
+.. code-block:: console
+
+   src/mytask/mytask.py run --logging DEBUG --local my-input-file.json
+
+
+The first argument is the command, of which the only option currently is `run`.
+
+- `--logging <LEVEL>` - configure the logging level of the task, one of DEBUG, INFO, WARN, ERROR, or CRITICAL
+- `--local` - sets several other flags to reasonable values for local testing, including `save-workdir`,
```
Comment (stark525) [inline docs/source/usage.rst:182]: `sets several other flags to reasonable values for local testing using a local payload file` ?

Something to capture that the payload itself is also in your project directory, whereas normally `input` is a remote destination, such as an S3 URI?

**Comment by stark525** [comment]
This is all fantastic 👍 

Not directly related to this PR, and this is my bias - but I have worked with a lot of people trying to Bring Your Own Container (BYOC), and stac-task is confusing with regard to how to migrate to use it, especially since it's "a wrapper". People think there should be a line like `# <---- RUN_YOUR_CONTAINER_HERE `."

I'm not sure that this is a use case immediately supported, or the best way to document it. This should probably be broken out into its own PR / discussion.

I think a lot of what's missing is the philosophy of how/why we've structured stac task this way and its advantages over doing some of this work (download images, create stac records manually, etc) in a wrapper like stac-task vs within a docker container itself. `PHILOSOPHY.md`? 😂

File: docs/source/usage.rst (hunk 3 above)
Comment (philvarner) [reply docs/source/usage.rst:162]: I made them consistent -- stac-task for the general library name, stactask for the package, STAC Task for the formal name.

**Pull request merged**